cadence/contracts/bridge/FlowEVMBridgeResolver.cdc

```diff
@@ -0,0 +1,156 @@
+import "MetadataViews"
+import "FungibleTokenMetadataViews"
+import "FungibleToken"
+import "NonFungibleToken"
+
+import "ICrossVMAsset"
+import "CrossVMToken"
+import "CrossVMNFT"
+import "FlowEVMBridgeUtils"
+
+/// This contract serves as a metadata resolver for shared views on NFTs and FungibleTokens bridged from EVM to Cadence.
+/// Upon bridging to Cadence, these tokens must have certain basic metadata views that cannot be retrieved at the time
+/// of bridging, such as MetadataViews.Display for NFTs and FungibleTokenMetadataViews.FTDisplay for FungibleTokens.
+///
+/// This contract then serves as a means of setting and potentiall updating base placeholder views for these bridged
+/// assets. Anyone wishing to consume the metadata from the source EVM contract should either resolve directly from the
+/// EVM contract or leverage the EVMBridgedMetadata view which retains the original token URI and/or contract URI (if
+/// implemented in the source contract).
+///
+access(all) contract FlowEVMBridgeResolver {
+
+    /******************
+        Entitlements
+    *******************/
+
+    access(all) entitlement Metadata
+
+    /*************
+        Fields
+    **************/
+
+    /// The views managed by this contract, indexed on the view type
+    access(all) let views: {Type: AnyStruct}
+
+    /********************
+        Path Constants
+    *********************/
+
+    /// The default storage path of the Admin resource
+    access(all) let AdminStoragePath: StoragePath
+
+    /*************
+        Events
+    **************/
+
+    /// Emitted when a view is added, removed, or updated.
+    /// added == true -> new | added == false -> removed | added == nil -> updated
+    access(all) event ViewUpdated(viewType: String, added: Bool?)
```

Comment:
Is there any other information about the view we could add, like a copy of the new view?

Reply:
Since this event is used for various views and they don't share a common interface, I'd need to create an event for each view (can't emit `AnyStruct`). Do you think that's alright? I suppose we could update to add events if we end up adding a new base bridged view, but I've been told it's not great to have such niche events.